src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -445,6 +444,22 @@ void Unit::Update(uint32 p_time)
 
     m_combatManager.Update(p_time);
 
+    if (_lastDamagedTargetGuid)
+        _lastDamagedTargetGuid = ObjectGuid::Empty;
```

Comment:
should the if check be removed here ? in the end the result is always ObjectGuid::Empty

Reply:
The purpose of resetting it is that extra attack spells direct casts add extra attacks on current target and not some dead / non-existant / last attacked unit. See Unit::AddExtraAttacks().
Imagine a situation where mob hits player1and then gets taunted by player2 30 yds away. Mob targets player2 and casts Thrash. It will trigger extra attacks on player1 unless `lastDamagedTargetGuid` is empty.

Reply:
I don't think you understoot what I mean.

Take `if (_lastDamagedTargetGuid)`, it can either be true or false:
- if it's true, `_lastDamagedTargetGuid = ObjectGuid::Empty;` will be called, meaning the stored value after the `if` check will be `ObjectGuid::Empty`
- if it's false, it means the stored value is already `ObjectGuid::Empty`

No matter if the `if` check evaluates to true or false, the value of `_lastDamagedTargetGuid` is always `ObjectGuid::Empty` at the end.

It's more of nitpicking, it might be that this PR is big enough to skip these small changes and focus instead on the big picture, my bad.

Reply:
Ah, sorry, didn't read carefully enough.

Yeah, I agree. Performance is pretty much unrelated here, so this check is unnecessary.